src/foam/nanos/auth/email/EmailVerificationWebAgent.js

```diff
@@ -94,20 +83,14 @@ foam.CLASS({
           translatedMsg = ts.getTranslation(local, getClassInfo().getId()+ ".EMAIL_VERIFIED_ERROR", this.EMAIL_VERIFIED_ERROR);
           message = translatedMsg + "<br>" + msg;
         } finally {
-          if ( getConfig() == null ) {
-            setConfig(EnvironmentConfigurationBuilder
-                .configuration()
-                .resources()
-                .resourceLoaders()
-                .add(new TypedResourceLoader("dao", new DAOResourceLoader(x, (String) user.getGroup())))
-                .and().and()
-                .build());
-          }
-    
+          EmailTemplateEngine templateEngine = (EmailTemplateEngine) x.get("templateEngine");
+          templateEngine.setX(x);
```

Comment:
Why do we need to setX()?

Reply:
I'm using context to find super email template

Reply:
When the service is created, if it is ContextAware, then NANOS will automatically call setX() on it.